src/utils/filter-utils.js

```diff
@@ -275,9 +374,11 @@ export function filterData(data, dataId, filters) {
  * @param {Object[]} data
  * @param {Object} filter
  * @param {number} i
+ * @param {field} field containing values to test data against. This is used only when
+ * testing timestamp filters
  * @returns {Boolean} - whether value falls in the range of the filter
  */
-export function isDataMatchFilter(data, filter, i) {
+export function isDataMatchFilter(data, filter, i, field) {
   const val = data[filter.fieldIdx];
```

Comment:
`filter.fieldIdx` should be one of the `fieldIdx` you are filtering based on. or simply use `field.tableFieldIndex - 1`

`const val = data[field.tableFieldIndex - 1];`